rspace/src/test/scala/coop/rchain/rspace/IStoreTests.scala

```diff
@@ -182,6 +183,54 @@ trait IStoreTests
       }
     }
   }
+
+  "pruneHistory" should "work on empty history" in withTestStore { store =>
+    store.pruneHistory(List.empty) shouldBe List.empty
+  }
+
+  it should "return unmodified history when nothing to prune" in withTestStore { store =>
+    forAll("gnat") { (gnat: GNAT[String, Pattern, String, StringsCaptor]) =>
+      val history = List(TrieUpdate(0, Insert, store.hashChannels(gnat.channels), gnat))
+      store.pruneHistory(history) shouldBe history
+    }
+  }
+
+  // TODO: Does this requirement make sense?
+  it should "remove all operations from history with the same hash when last operation is delete" in withTestStore {
+    store =>
+      forAll("gnat1", "gnat2") {
+        (gnat1: GNAT[String, Pattern, String, StringsCaptor],
+         gnat2: GNAT[String, Pattern, String, StringsCaptor]) =>
+          val gnat1Ops = List(TrieUpdate(0, Insert, store.hashChannels(gnat1.channels), gnat1),
+                              TrieUpdate(1, Delete, store.hashChannels(gnat1.channels), gnat1))
+          val gnat2Ops = List(TrieUpdate(2, Insert, store.hashChannels(gnat2.channels), gnat2))
+          val history  = gnat1Ops ++ gnat2Ops
```

Comment:
Seems like we'd want to break out a separate test here where you are only doing operations on a single key:
```scala
val gnatOps = List(TrieUpdate(0, Insert, store.hashChannels(gnat1.channels), <something>),
                   TrieUpdate(1, Insert, store.hashChannels(gnat1.channels), <something>),
                   TrieUpdate(2, Insert, store.hashChannels(gnat1.channels), <something>),
                   TrieUpdate(3, Delete, store.hashChannels(gnat1.channels), <something>))

store.pruneHistory(gnatOps) should be empty

and again with the list jumbled...

Reply:
That was the initial idea, but then I got carried away with generators. Will add

Reply:
Done